palace/utils/geodata.cpp

```diff
@@ -689,43 +682,52 @@ BoundingBox BoundingBoxFromPointCloud(MPI_Comm comm, std::vector<Eigen::Vector3d
   MPI_Gatherv(vertices.data(), 3 * num_vertices, MPI_DOUBLE, collected_vertices.data(),
               recv_counts.data(), displacements.data(), MPI_DOUBLE, dominant_rank, comm);
 
-  BoundingBox box;
+  // Deduplicate vertices. Given floating point precision, need a tolerance.
   if (dominant_rank == Mpi::Rank(comm))
   {
-    // dominant_rank now has a fully stocked vector of vertices. All other ranks will wait
-    // for results. Deduplicate vertices. Given floating point precision, need a tolerance.
     auto vertex_equality = [](const auto &x, const auto &y)
     {
       constexpr double tolerance = 10.0 * std::numeric_limits<double>::epsilon();
       return std::abs(x[0] - y[0]) < tolerance && std::abs(x[1] - y[1]) < tolerance &&
              std::abs(x[2] - y[2]) < tolerance;
     };
-
     vertices = std::move(collected_vertices);
     std::sort(vertices.begin(), vertices.end(), EigenLE);
     vertices.erase(std::unique(vertices.begin(), vertices.end(), vertex_equality),
                    vertices.end());
-    MFEM_VERIFY(vertices.size() >= 4,
-                "A bounding box requires a minimum of four vertices for this algorithm!");
+  }
+  else
+  {
+    vertices.clear();
+  }
+
+  return dominant_rank;
+}
 
+// Calculates a bounding box from a point cloud, result is broadcast across all processes.
+BoundingBox BoundingBoxFromPointCloud(MPI_Comm comm, std::vector<Eigen::Vector3d> &vertices,
```

Comment:
`const` on the vector

Reply:
Fixed in [fa13b91](https://github.com/awslabs/palace/pull/75/commits/fa13b9127936f410ec4f4ddb599c29abac62e848).